SCF.py

```diff
@@ -159,20 +134,21 @@ def form_density_matrix(mol_, mo_coeffs_):
 
     nelec = mol_.nelec[0]  # Number of occupied orbitals
     num_aos = mol_.nao  # Number of atomic orbitals, dimensions of the mats
-    Duv = np.zeros(mol_.nao, mol_.nao, dtype=np.double)
-
-    """
-    Replace with your implementation
+    Duv = np.zeros((mol_.nao, mol_.nao), dtype=np.double)
 
-    This involves basically a computation of each density matrix element
-    that is a sum over the produces of the mo_coeffs.
+    for i in range(nelec):

```

Comment:
nelec should be the inner most loop variable here, not the outer most. You should always loop over the matrix elements to compute first.